### ISSUE TITLE
Kodi crash and device crashes

### ISSUE THREAD

**Issue by gmjh123** [matthuisman/slyguy.addons#830]
Hi Matt. Firstly I found today that paramount plus and Max would not play any audio at all. I reverted from inputstream adaptive 21.5 to 21.4 and then the audio worked. I think that issue has been discussed before. My much bigger problem is that when playing files from the Hulu add-on, the file starts to play, freezes, and then the whole device crashes {Onn 4k pro Android TV based) and restarts. Bizarrely when the device restarts and I open Kodi, all settings on Kodi have been reverted to default. It hasn't reverted everything to default, all of my add-ons are still there and all of my shortcuts, but all Kodi settings such as the settings of media player service, system, etc are all wiped and returned to default. 

I think you're probably going to ask for a full Kodi debug log, but I'm not sure how it can log this because once the entire device crashes and I ask for a debug log it's only going to show a log from after it restarted it again. Isn't it? Thanks very much

**Comment by matthuisman** [comment]
@gmjh123 best to stay on 21.5.0 and just force 2 channel audio for now (can do that via addon -> settings -> player -> codecs -> max audio channels -> 2)

Once done that, then please test hulu etc. Make sure you running latest slyguy common as well as I did a hulu fix recently :)
And 21.5.0 also had some crashing fixes. But does have the 5.1 audio bug

**Comment by gmjh123** [comment]
Ok thanks. But as I said elsewhere, it's probably much easier just to play
anything Hulu on the Disney plus add on which doesn't seem to have any
issues at all.

I'd also prefer just running the older version of inputstream adaptive on
Max and paramount add-ons to get 5.1 or Atmos. It's better than silence or
slightly crappy stereo 🙂

**Comment by matthuisman** [comment]
Not sure where "elsewhere" is either?
its not my addon issue, its inputstream.adaptive issues
Out of my hands

**Comment by gmjh123** [comment]
Elsewhere is the email I sent you this morning 🙂

**Comment by matthuisman** [comment]
oh, its different names between here and there so didnt connect the two :)

So your now using 21.4.10 IA?
The hulu issues sounds like this: https://github.com/xbmc/inputstream.adaptive/issues/1530
Which the dev found was due to odd format of their widevine default kids
2 days ago i update slyguy common proxy to fix those on the fly which should fix that above issue

Have you tried the Hulu add-on with [script.module.slyguy -> v0.84.41](https://github.com/matthuisman/slyguy.addons/commit/7a81c1bcd7b3a0cf50dfbde9fab99e104da4b313) or newer?

**Comment by gmjh123** [comment]
To be honest I really don't want to test Hulu anymore because, as I said,
every time Hulu freezes and my entire device restarts, Kodi wipes all of my
settings which take me about 10 to 15 minutes to set up again.

I might try it again in a couple of months

**Comment by matthuisman** [comment]
do the kodi settings themselves get wiped when it crashes??

**Comment by gmjh123** [comment]
Yes, all settings on Kodi get reset to default and at Standard not Expert
settings. It takes me a long time to remember what my preferred settings
were and then to change them all back to what I want. It doesn't wipe any
add-ons, it doesn't wipe any add-on preferences, and it doesn't wipe any of
my shortcuts. But everything else on Kodi gets wiped and put back to
default.

**Comment by matthuisman** [comment]
ok thanks. so yeh - seems kodi corrupts its guisettings.xml and probably something else.
You could actually backup that file yourself and just copy it back over in-case of emergencies
I do this when setting up new clients. 

userdata / 
  guisettings.xml 
  advancedsettings.xml 
  passwords.xml
  sources.xml
  
Pretty much all you should need.


Anway, last thing - can you advise when you last tried the Hulu addon?
Do you think it was before you had my https://github.com/matthuisman/slyguy.addons/commit/7a81c1bcd7b3a0cf50dfbde9fab99e104da4b313 update which went live 2 days ago. That works around the Inpustream adaptive bug (I have a lot of workarounds for that thing haha)

**Comment by gmjh123** [comment]
Yes I should use a backup file, but I don't think I can save it to my
Android device, I think it would have to be on a USB. I haven't tried Hulu
recently up until today, so I think that would have been after your update.
I then got a new subscription for Disney plus, which was offered by my ISP
for almost nothing. So thankfully I don't have to use the Hulu add-on at
all.

**Comment by matthuisman** [comment]
ok thank you.

What bit of content did you try on Hulu today that killed it?
Ill try with the same movie / episode

**Comment by gmjh123** [comment]
I think one of them was an episode of little fires everywhere. But I think
the big crash thing was before I changed input stream adaptive to the
earlier version.

**Comment by dshorrosh** [comment]
Hi Matt - I've tried to follow this and related threads about the Hulu freezing issue (mainly due to (inputstream.adaptive) and the way I read things, the issue should be resolved.  Hulu LIVE TV is still freezing with Kodi v21.1 on both Win10 and RP boxes (VOD plays fine).  Sorry to be so thick but are there specific steps I need to take for this to work?  I can give fresh log if needed.

**Comment by gmjh123** [comment]
> Hi Matt - I've tried to follow this and related threads about the Hulu freezing issue (mainly due to inputstream.adaptive) and the way I read things, the issue should be resolved. Hulu LIVE TV is still freezing with Kodi v21.1 on both Win10 and RP boxes (VOD plays fine). Sorry to be so thick but are there specific steps I need to take for this to work? I can give fresh log if needed.

I'm the op. I still can't play anything on Hulu that involves the Hulu green and black Hulu ident at the start. Hulu uses a different resolution and a different audio for this little intro thing, and it completely screws up the stream on the add-on. I suspect there's no fix for this, as it's been happening for about 2 years. Luckily watching all Hulu content on the Disney add on (which doesn't use the green and black Hulu intro) is flawless and much better quality to.

**Comment by gmjh123** [comment]
> > I'm the op. I still can't play anything on Hulu that involves the Hulu green and black Hulu ident at the start. Hulu uses a different resolution and a different audio for this little intro thing, and it completely screws up the stream on the add-on. I suspect there's no fix for this, as it's been happening for about 2 years. Luckily watching all Hulu content on the Disney add on (which doesn't use the green and black Hulu intro) is flawless and much better quality to.
> 
> OP - Thanks, I'll explore how to see the Hulu live channels via Disney+ addon, if I'm understanding correctly. FWIW I'm using Hulu for what used to be 'cable channels' with IPTV Merge and NextPVR (OTA channels).

Oh okay I didn't realize you are almost exclusively using it for Hulu live channels. I'm pretty sure the Disney add-on and the official Disney app doesn't have any Hulu live channels.

**Comment by dshorrosh** [comment]
> Oh okay I didn't realize you are almost exclusively using it for Hulu live channels. I'm pretty sure the Disney add-on and the official Disney app doesn't have any Hulu live channels.

Ah, got it.  Bottom line, LIVE TV still derps with no known fix.  Thanks for explaining and sounds like I've got some new research to do.

Thanks @matthuisman for the work you and your team do on the addons.

**Comment by matthuisman** [comment]
@dshorrosh I think https://github.com/xbmc/inputstream.adaptive/issues/1426 is what your looking for

**Comment by dshorrosh** [comment]
Thanks for replying @matthuisman - I'm new to GitHub and the way I read https://github.com/xbmc/inputstream.adaptive/issues/1426 live channels are still broken for the Hulu addon.  Sorry for the hand-holding but let me know if I'm misunderstanding (and what files to download / override if there is a fix).
